### PR TITLE
Shader testing framework

### PR DESCRIPTION
**Pull request by alexd2580**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,353 @@
+//! This crate provides macros for writing simple vulkan compute shader tests
+//! using the tomaka/vulkano library.
```
Comment (svenstaro) [inline src/lib.rs]: Link to it here.

File: src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+//!
+//! ## About
+//!
+//! A core problem of developing shaders is the rather difficult environment in which they are
+//! executed. Even simple things can go wrong and cost the developer a lot of time to fix.
+//! This crate aims at providing a simple-to-use environment for writing vulkan compute shader tests.
+//! It uses the vulkano rust-vulkan bindings end exports macros for a fast implementation of tests.
+//! These macros mostly generate vulkano boilerplate instantiation code. The interface to the
+//! shader are cpu accessible buffers which you can read and write at will and a function for
```
Comment (svenstaro) [inline src/lib.rs]: > CPU

File: src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+                       &BufferUsage::all(),
+                       Some($queue.family()))
+                   .expect("Failed to create a cpu accessible buffer.")
+        }
+    })
+}
+
+/// This macro is the core of the shader-testing framework.
+/// It generates code for initializing the vulkano environment,
+/// it allocates CPU-side buffers, it compiles the shader,
```
Comment (svenstaro) [inline src/lib.rs]: > CPU accessible

File: src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+///     shader_path: "tests/shaders/example.comp",
+///     workgroup_count: [100, 100, 1],
+///     buffers: {
+///        data: [u32; total_num_invocations],
+///        result: [u32; total_num_invocations]
+///     },
+///     execution_command: execute_shader
+/// }
+///
+/// // II. Fill your buffers with input data. The buffers are binded to the
```
Comment (svenstaro) [inline src/lib.rs]: > bound

File: tests/random.rs [hunk 5]
```diff
@@ -0,0 +1,71 @@
+//! This crate provides macros for writing simple vulkan compute shader tests
```
Comment (svenstaro) [inline tests/random.rs]: This should be a proper module documentation. What are you testing here?

File: src/lib.rs (hunk 1 above)
Comment (alexd2580) [inline src/lib.rs]: Fixed in 7a7f036.

File: src/lib.rs (hunk 2 above)
Comment (alexd2580) [inline src/lib.rs]: Fixed in 7a7f036.

File: src/lib.rs (hunk 3 above)
Comment (alexd2580) [inline src/lib.rs]: Fixed in 7a7f036.

File: src/lib.rs (hunk 4 above)
Comment (alexd2580) [inline src/lib.rs]: Fixed in 7a7f036.

File: tests/random.rs (hunk 5 above)
Comment (alexd2580) [inline tests/random.rs]: Fixed in 7a7f036.

**Comment by alexd2580** [comment]
@svenstaro The things you have mentioned are corrected.

**Pull request merged**